src/index.js

```diff
@@ -183,7 +186,7 @@ const embedPublicKey = async (publicKey, entry) => {
  */
 const extractPublicKey = (peerId, entry) => {
   if (!entry || !peerId) {
-    const error = 'one or more of the provided parameters are not defined'
+    const error = new Error('one or more of the provided parameters are not defined')
```

Comment:
Should we do the `new Error` only on throw?

Same applies on another case below

Reply:
There won't be a stack trace in the logs if we just log the message.

Reply:
I usually prefer to have clean logs and with the ERR CODE find and replicate the issue. But, I understand your point as well

Reply:
If there's only one place in the entire codebase a certain error code is emitted, and it's also only invoked from one place in the entire codebase I'd agree, but that's rarely the case so more verbose logs can only help.